mautic/mautic
# TPROD-284 Refactor CookieHelper to use Symfony Cookie.

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [ x ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ x ]
| Automated tests included?              | [ x ] 



#### Description:
Refactor CookieHelper to use Symfony Cookie. See https://mautic.atlassian.net/browse/TPROD-284
BC break: methods calls are now strictly typed.


#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. The cookie is set when the page is hit (in incognito)
